consensus/ibft/bridge.go

```diff
@@ -140,6 +140,13 @@ func (b *BridgeMechanism) consumeStateTransactionsHook(numberParam interface{})
 		b.bridge.StateSync().Consume(tx)
 	}
 
+	// On every epoch end, create the checkpoint
+	if b.ibft.IsLastOfEpoch(block.Number()) {
+		if err := b.bridge.StartNewCheckpoint(b.ibft.epochSize); err != nil {
+			return err
+		}
+	}
```

Comment:
@Kourin1996 

As far as I can see, this is triggered in the `InsertBlock` hook for the Bridge mechanism, which is called during _both_ commit state and sync state. Why create checkpoints during sync state?

Reply:
Yes, you are right, I did not know that this hook is called from sync state. 
My first idea was to create a new hook which would be called from insertBlock. Buy @Kourin1996 suggested we should stick with the one we have. 
@Kourin1996 How do you think we should handle this?

Reply:
Why not use `AcceptStateLogHook` or some hooks called in runAcceptState. Actually don't have to call this immediately after you insert block

Reply:
I see that AcceptStateLogHook is used for logging validators for POA and POS, and updating validators for Bridge. I do not think that starting checkpoint should happen in a method called updateValidatorsHook. It's not intuitive to start a checkpoint from a hook called AcceptStateLogHook. 

Also somewhere in the future, we would maybe want to use this hook in some other place and we would not be able because of checkpointing process.

Reply:
Yes, this is current problem. The hook Naming convention is not good because it's depending on extension. I meant why you call CreateCheckpoint in runAcceptState? I was just talking about the time when you call it.

Reply:
I do not see any reason why we would not call CreateCheckpoint from runAcceptState.

Reply:
then why don't you move?

Reply:
We can create a checkpoint from this hook, then refactor bridge hooks in the main bridge branch?

Reply:
Only thing you need to care about is that latest block height must be greater than the end block height of next checkpoint. What's the difference between calling CreateCheckPoint on InsertBlockHook and hook called from runAcceptState?

Reply:
Fixed bf961fd879108c34d17d4f6808befcf3b6ea5812. @Kourin1996 check this commit please. I have done manual testing to see if everything works

Reply:
+1 on what @Kourin1996 said, everything was on point

Reply:
@Aleksao998 
Why are you calling the checkpoint start from the `updateValidatorsHook`?

Reply:
Fixed df8a837dd41fadcdf9bd23709ed0b2e78cd65aed

Reply:
Closing as it's been fixed in [ec122b7](https://github.com/0xPolygon/polygon-edge/pull/466/commits/ec122b7415e1b37c8cd7c55a8cd2700a006ab1e7) and subsequent commits